Part_2/Search/344.cpp

```diff
@@ -0,0 +1,112 @@
+#include <iostream>
+#include <set>
+
+class Matrix
+{
+    double** matrix;
+    int length;
+    int width;
+
+public:
+    Matrix(int a_length, int a_width)
+    {
+        length = a_length;
+        width = a_width;
+
+        matrix = new double* [length];
+        
+        for (int i = 0; i < length; i++)
+        {
+            matrix[i] = new double[width];
+        }
+    }
+
+    void input() 
+    {
+        std::cout << "Type the values in the matrix: ";
+
+        for (int i = 0; i < length; i++) 
+        {
+            std::cout << "Line " << i << ": \n";
+            for (int j = 0; j < width; j++)
+            {
+                std::cin >> matrix[i][j];
+            }
+        }
+    }
+
+    void print()
+    {
+        std::cout << "\nPrint Matrix: \n";
+        for (int i = 0; i < length; i++)
+        {
+            for (int j = 0; j < width; j++)
+            {
+                std::cout << matrix[i][j] << " ";
+            }
+            std::cout << "\n";
+        }
+        std::cout << "\n";
+    }
+
+    int numberOfValues() 
+    {
+        int totalNumberOfValues = 0;
+
+        std::set<double> setOfValues;
+        std::set<double>::iterator it;
+
+        for (int i = 0; i < length; i++)
+        {
+            for (int j = 0; j < width; j++)
+            {
+                if (setOfValues.find(matrix[i][j]) == setOfValues.end())
```

Comment:
We can use the `insert` method and check the success in the return value of `insert`. https://cplusplus.com/reference/set/set/insert/